protocol/messenger_handler.go

```diff
@@ -2199,6 +2204,11 @@ func (m *Messenger) handleChatMessage(state *ReceivedMessageState, forceSeen boo
 
 	contact := state.CurrentMessageState.Contact
 
+	if chat.ChatType == ChatTypeOneToOne && contact.Dismissed() {
+		m.logger.Info("ignoring one on one messages dismissed contacts")
+		return nil
+	}
+
```

Comment:
There is a `isMessageAllowedFrom` here:
https://github.com/status-im/status-go/blob/5825d1cba51ab86397c50928832c467db2438512/protocol/messenger_handler.go#L1128-L1134

Looks to me like `Dismissed` check should be part of it.

And there, looks like we should "deactive" the chat when dismissing the contact, then this function will work properly, without any custom checks for `contact.Dismissed()`:
https://github.com/status-im/status-go/blob/5825d1cba51ab86397c50928832c467db2438512/protocol/messenger_handler.go#L2951-L2958

Reply:
I did try to modify the `isMessageAllowedFrom` function and it had a bad side-effect. IIRC, it made it impossible to receive some other messages from them. I can try again to remember what was the issue, but what it made me realize is that function is badly named

Reply:
> What I'm saying is that maybe we should set chat.Active = false when dismissing the contact 🤔

Yeah I guess it makes sense. Let's try